docs/user-guide/content-type.md

```diff
@@ -0,0 +1,232 @@
+# Content Types
+
+Machine learning models generally expect their inputs to be passed down as a
+particular Python type.
+Most commonly, this type ranges from _"general purpose"_ NumPy arrays or Pandas
+DataFrames to more granular definitions, like `datetime` objects, `Pillow`
+images, etc.
+Unfortunately, the definition of the [V2 Inference
+Protocol](https://kserve.github.io/website/modelserving/inference_api/) doesn't
+cover any of the specific use cases.
+This protocol can be thought of a wider _"lower level"_ spec, which only
+defines what fields should a payload have.
+
+To account for this gap, MLServer introduces support for **content types**,
+which offer a way to let MLServer know how it should _"decode"_ V2-compatible
+payloads.
+When shaped in the right way, these payloads should _"encode"_ all the
+information required to extract the higher level Python type that will be
+required for a model.
+
+To illustrate the above, we can think of a Scikit-Learn pipeline, which takes
+in a Pandas DataFrame and returns a NumPy Array.
+Without the use of **content types**, the V2 payload itself would probably lack
+information about how should this payload be treated by MLServer.
+Likewise, the Scikit-Learn pipeline wouldn't know how to treat a raw V2
+payload.
+In this scenario, the use of content types allows us to specify information on
+what's the actual _"higher level"_ information encoded within the V2 protocol
+payloads.
+
+![Content Types](../assets/content-type.svg)
+
+## Usage
+
+To let MLServer know that a particular payload must be decoded / encoded as a
+different Python data type (e.g. NumPy Array, Pandas DataFrame, etc.), you can
+specifity it through the `content_type` field of the `parameters` section of
+your request.
+
+As an example, we can consider the following dataframe, containing two columns:
+Age and First Name.
+
+| First Name | Age |
+| ---------- | --- |
+| Joanne     | 34  |
+| Michael    | 22  |
+
+This table, could be specified in the V2 protocol as the following payload, where we declare that:
+
+- The whole set of inputs should be decoded as a Pandas Dataframe (i.e. setting
+  the content type as `pd`).
+- The Age column should be decoded as a UTF-8 string (i.e. setting the content
```

Comment:
Did you mean "The First name column"? The Age column is decoded as an integer

Reply:
Good catch! :+1: